File: Mansooj/python/tests/join_tests/test_init_join.py
# -*- coding: utf-8 -*-
"""
JoinEngine — Initialization Unit Test
-------------------------------------
Verifies creation and initialization of Mansooj JoinEngine via CTypes binding.
"""

import unittest
import sys
import os


sys.path.append(os.path.join(os.path.dirname(__file__), "../../"))


from mansooj_join.mansooj_join import (
    create_default_join_config,
    create_join_engine,
    init_join_engine,
)
class TestJoinEngineInit(unittest.TestCase):
    def test_join_engine_init(self):
        print("\nRunning Mansooj JoinEngine init test")

        cfg = create_default_join_config()
        self.assertIsNotNone(cfg, "JoinConfig creation failed")
        print(f"→ Config created (encoding={cfg.encoding})")

        eng = create_join_engine(cfg)
        self.assertIsNotNone(eng, "JoinEngine creation failed")
        print("→ Engine created")

        init_join_engine(eng, cfg)
        print("→ Engine initialized")

        # Basic validation
        self.assertTrue(hasattr(cfg, "encoding"))
        self.assertTrue(hasattr(eng, "contents") or hasattr(eng, "value"))
        print("✅ JoinEngine init verification passed")

if __name__ == "__main__":
    unittest.main(verbosity=2)

File: Mansooj/php/tests/join_tests/test_init_join.php
<?php
/**
 * Mansooj JoinEngine — Initialization Unit Test
 * ---------------------------------------------
 */

require_once __DIR__ . '/../../mansooj_join/mansooj_join.php';

echo "\nRunning Mansooj JoinEngine init test\n";

try {
    $cfg = create_default_join_config();
    if (!$cfg) throw new Exception("JoinConfig creation failed");
    echo "→ Config created (encoding={$cfg->encoding})\n";

    $eng = create_join_engine($cfg);
    if (!$eng) throw new Exception("JoinEngine creation failed");
    echo "→ Engine created\n";

    init_join_engine($eng, $cfg);
    echo "→ Engine initialized\n";

    if (!property_exists($cfg, "encoding"))
        throw new Exception("Missing encoding field");
    if (!property_exists($eng, "cfg"))
        throw new Exception("Engine missing cfg link");

    echo "JoinEngine init verification passed\n";
} catch (Exception $e) {
    echo "Error: {$e->getMessage()}\n";
    exit(1);
}
